pkg/kubelet/dockertools/manager.go

```diff
@@ -573,13 +573,21 @@ func makeEnvList(envs []kubecontainer.EnvVar) (result []string) {
 // can be understood by docker.
 // Each element in the string is in the form of:
 // '<HostPath>:<ContainerPath>', or
-// '<HostPath>:<ContainerPath>:ro', if the path is read only.
-func makeMountBindings(mounts []kubecontainer.Mount) (result []string) {
+// '<HostPath>:<ContainerPath>:ro', if the path is read only, and/or
+// '<HostPath>:<ContainerPath>:Z', if the volume requires SELinux
+// relabling and the pod provides an SELinux label
+func makeMountBindings(mounts []kubecontainer.Mount, podHasSELinuxLabel bool) (result []string) {
 	for _, m := range mounts {
 		bind := fmt.Sprintf("%s:%s", m.HostPath, m.ContainerPath)
 		if m.ReadOnly {
 			bind += ":ro"
 		}
+		// If the pod does not provide an SELinux context
+		// relabeling will restrict access to the volume the
+		// container which mounts it last
+		if m.SELinuxRelabel && podHasSELinuxLabel {
```

Comment:
Also, if the pod uses the hostPid namespace, we will need to chcon the volume in the docker manager instead of relying on docker.  I think we should extract the `ChconRunner` interface into its own package.  I'm going to add siblings to it.  Following the pattern of mount, I was thinking: `pkg/chmod/chmod.go`:

```go
package chmod

type Interface interface {
  Chmod(path, mode string) error
}
```

Thoughts about that?